src/main/org/firebirdsql/gds/ng/wire/WireConnection.java

```diff
@@ -250,7 +275,7 @@ public final C identify() throws SQLException {
 
             xdrOut.writeInt(op_connect);
             xdrOut.writeInt(op_attach);
-            xdrOut.writeInt(CONNECT_VERSION2);
+            xdrOut.writeInt(CONNECT_VERSION3);
```

Comment:
Yes, it works. Historically, FB server never checks the connect version. Version 3 is checked by FB3 to ensure that CNCT_* strings (inside op_connect) are UTF8 encoded, priorly their encoding was unknown (fbclient used system encoding for them).